### ISSUE TITLE
Kitsu updating bug

### ISSUE THREAD

**Issue by TSedlar** [TSedlar/anitrack#17]
The Kitsu PATCH protocol does not work for library modifications.

I have contacted the developer of the Kitsu API to get more information on this.

There is a workaround to modify your library and grab the Bearer AUTH that Kitsu will generate, but it only works for 24 hours, thus not being stable in this.

The public Bearer auth in the Kitsu API works for everything else, just not modification.

Auth: 54d7307928f63414defd96399fc31ba847961ceaecef3a5fd93144e960c0e151

Response:
```
403 Forbidden: You don’t have permission to update the library entry.
```

**Comment by astraldragon** [comment]
Maybe you have the authorization confused?  You need to auth against their oauth endpoint with users username and password to get the oauth token you would send in subsequent calls to update their library.

**Comment by TSedlar** [comment]
I don't believe they have that implemented currently. The official node API has this listed for authentication:

![](https://i.imgur.com/SZ2JjY8.png)

and the documentation lists the following:

![](https://i.imgur.com/gtL5o29.png)

so the key in the documentation should work.

**Comment by astraldragon** [comment]
Right but to modify someone's library you have to be authenticated as that person.  If that wasn't the case you'd be able to modify everyone's libraries :)

**Comment by TSedlar** [comment]
@astraldragon 

I've also tried using the [Basic authentication](https://github.com/TSedlar/anitrack/blob/master/src/shared/app/service/Kitsu.js#L11-L13)

Is there another way to go about it? Because that works fine through MyAnimeList

**Comment by astraldragon** [comment]
MyAnimeList is an entirely different method of authentication.  Kitsu uses oauth so as far as I know basic auth will not work.  You could use something like https://github.com/lelylan/simple-oauth2 or https://github.com/mulesoft/js-client-oauth2 to get yourself up and running quick

**Comment by TSedlar** [comment]
Thanks @astraldragon!

Fixed in 901adf7
